controller/gif-controller.js

```diff
@@ -0,0 +1,95 @@
+const db = require("../db/db");
+const cloudinary = require("cloudinary");
+require("dotenv").config();
+
+const createGif = async (req, res) => {
+  // get the token details from the authmiddleware
+  const user = req.user;
+  const { base64EncodedGif, title } = req.body;
+
+  // configure cloudinary
+  cloudinary.config({
+    cloud_name: process.env.CLOUD_NAME,
+    api_key: process.env.CLOUD_API_KEY,
+    api_secret: process.env.CLOUD_API_KEY_SECRET,
+  });
+  try {
+    // get user id from the authenticated user details
+    const userId = user.id;
+
+    const uploadResult = await cloudinary.v2.uploader.upload(base64EncodedGif, {
+      resource_type: "auto",
+    });
+
+    // Create the gif table if it doesn't exist
+    const createTableQuery = `
+          CREATE TABLE IF NOT EXISTS gifs (
+            id SERIAL PRIMARY KEY,
+            url TEXT,
+            title TEXT,
+            user_id INT,
+            created_on TIMESTAMP DEFAULT NOW()
+          );
+        `;
+    await db.query(createTableQuery);
```

Comment:
Why Are We Doing This Here?

Reply:
have set another file to create dummy data